src/main/flight/mixer.c

```diff
@@ -661,6 +661,130 @@ void StopPwmAllMotors()
 }
 
 #ifndef USE_QUAD_MIXER_ONLY
+static int16_t decidegreeToServoPulse(uint8_t servo_index, int16_t angle_deci_degree)
+{
+    // normalize angle to range -180 to 180 (degree)
+    while (angle_deci_degree > 1800)
+        angle_deci_degree -= 3600;
+    while (angle_deci_degree < -1800)
+        angle_deci_degree += 3600;
+
+    //remap input value (RX limit) to output value (Servo limit), also take into account eventual non-linearity of the two half range
+    if (angle_deci_degree > 0) {
+        angle_deci_degree = scaleRange(angle_deci_degree, 0, servoConf[servo_index].angleAtMax * 10, 0, servoConf[servo_index].max);
+    } else {
+        angle_deci_degree = scaleRange(angle_deci_degree, 0, servoConf[servo_index].angleAtMin * 10, 0, servoConf[servo_index].min);
+    }
+
+    return angle_deci_degree;
+}
+
+int16_t getMix(uint8_t rule_index)
+{
+
+    static int16_t currentOutput[MAX_SERVO_RULES];
+
+    uint8_t servo_index = currentServoMixer[rule_index].targetChannel;
+
+    // consider rule only if no box assigned or box is active
+    if (!currentServoMixer[rule_index].box == 0 && !IS_RC_MODE_ACTIVE(BOXSERVO1 + currentServoMixer[rule_index].box - 1)) {
```

Comment:
this is `(!currentServoMixer[rule_index].box) == 0` due to operator precedence. The result is correct, but the calculation is confusing. `currentServoMixer[rule_index].box) != 0` would be much easier to understand (`currentServoMixer[rule_index].box` is also common C idiom .. )